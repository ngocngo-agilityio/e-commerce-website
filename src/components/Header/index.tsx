// Libs
import { Box, Container, HStack, Show, Hide, Flex } from '@chakra-ui/react';

// Assets
import { HamburgerIcon } from '@assets';

// APIs
import { getCartItems } from '@apis';

// Auth configs
import { auth } from '@configs';

// Components
import { Logo, Navigation, ShoppingCart, UserProfile } from '@components';

const Header = async (): Promise<JSX.Element> => {
  const session = await auth();

  const userId = session?.user?.id || '';

  // Get my cart
  const { data: myCart } = await getCartItems(userId);

  const { cartItems = [] } = myCart || {};
  const cartItemQuantity = cartItems.length || 0;

  return (
    <Container
      as="header"
      display="flex"
      flexDir="column"
      gap={2}
      pt="34px"
      pb={{ base: 5, md: 4 }}
    >
      <Flex alignContent="center" justifyContent="space-between">
        <HStack spacing={{ md: '100px', lg: '150px' }} wrap="wrap">
          <Logo />
          <Show above="md">
            <Navigation />
          </Show>
        </HStack>

        <HStack spacing="25px" alignContent="center">
          <UserProfile />
          <ShoppingCart cartItemQuantity={cartItemQuantity} />

          <Box cursor="pointer">
            <HamburgerIcon />
          </Box>
        </HStack>
      </Flex>

      <Hide above="md">
        <Navigation />
      </Hide>
    </Container>
  );
};

export default Header;
