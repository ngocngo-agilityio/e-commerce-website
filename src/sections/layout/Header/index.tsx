// Libs
import {
  Box,
  Container,
  HStack,
  Show,
  Hide,
  Flex,
  Skeleton,
} from '@chakra-ui/react';
import { Suspense } from 'react';

// Assets
import { HamburgerIcon } from '@assets';

// Components
import { Logo, Navigation, UserProfile } from '@components';

// Sections
import { ShoppingCart } from '@sections';

const Header = (): JSX.Element => (
  <Container
    as="header"
    display="flex"
    flexDir="column"
    gap={2}
    pt="34px"
    pb={{ base: 5, md: 4 }}
  >
    <Flex alignContent="center" justifyContent="space-between">
      <HStack spacing={{ md: '100px', lg: '150px' }} wrap="wrap">
        <Logo />
        <Show above="md">
          <Navigation />
        </Show>
      </HStack>

      <HStack spacing="25px" alignContent="center">
        <UserProfile />
        <Suspense fallback={<Skeleton w={6} h={6} />}>
          <ShoppingCart />
        </Suspense>

        <Box cursor="pointer">
          <HamburgerIcon />
        </Box>
      </HStack>
    </Flex>

    <Hide above="md">
      <Navigation />
    </Hide>
  </Container>
);
export default Header;
