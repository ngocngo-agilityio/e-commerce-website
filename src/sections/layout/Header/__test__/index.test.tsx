// Libs
import { render } from '@testing-library/react';

// Components
import Header from '../';

jest.mock('@apis', () => ({
  ...jest.requireActual('@apis'),
  getCartItems: jest.fn(),
}));

jest.mock('@sections', () => ({
  ShoppingCart: () => <div>ShoppingCart</div>,
}));

jest.mock('@components', () => ({
  Logo: () => <div>Logo</div>,
  Navigation: () => <div>Navigation</div>,
  UserProfile: () => <div>UserProfile</div>,
}));

describe('Header component', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  test('should match snapshot for Header', async () => {
    const { container } = render(<Header />);

    expect(container).toMatchSnapshot();
  });
});
