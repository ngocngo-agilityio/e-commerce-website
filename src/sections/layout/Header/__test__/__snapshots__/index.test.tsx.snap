// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should match snapshot for Header 1`] = `
<div>
  <header
    class="chakra-container css-9uj2mi"
  >
    <div
      class="css-4kyngj"
    >
      <div
        class="chakra-stack css-1mra35c"
      >
        <div>
          Logo
        </div>
      </div>
      <div
        class="chakra-stack css-erq1zg"
      >
        <div>
          UserProfile
        </div>
        <div>
          ShoppingCart
        </div>
        <div
          class="css-4g6ai3"
        >
          <svg
            class="chakra-icon css-1t7rsys"
            focusable="false"
            viewBox="0 0 24 25"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3 12.4761H21"
              stroke="#1D1D1D"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
            <path
              d="M3 6.47607H21"
              stroke="#1D1D1D"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
            <path
              d="M3 18.4761H21"
              stroke="#1D1D1D"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </div>
      </div>
    </div>
    <div>
      Navigation
    </div>
  </header>
</div>
`;
